Fixed Shader not working in Version 1.20.X

TheIdealMatrix opened the pull request:
The Tooltip Shader has been moved in 1.20.X (from `position_color` to `rendertype_gui`) - Fixes #4 

- I've changed the File Names & also cleaned the Code a bit.
- Removed the "Long compat" mode option, because it can simply be enabled the whole time (didn't see any line at the bottom).
- Changed the README accordingly.
- Made the Lore longer to also get removed with GUI Scale 1 or larger monitors.
- Added /give Command to README.

Everything works as expected in Version 1.20.1 as tested by me (probably also 1.20).

TheIdealMatrix committed:
Fixed Shader not working in Version 1.20.X

kelllllen commented:
I don't know much about core shaders, but I saw this one runs on GLSL 440 core, which doesn't work with my Apple M2. I also saw that all the minecraft core shaders run on 150. Is it possible to downgrade this to 150?

TheIdealMatrix committed:
Downgraded Version to 150 core

TheIdealMatrix commented:
> I don't know much about core shaders, but I saw this one runs on GLSL 440 core, which doesn't work with my Apple M2. I also saw that all the minecraft core shaders run on 150. Is it possible to downgrade this to 150?

Seems to work for me. I've changed it.

kelllllen commented:
Thanks! Unfortunately, it didn't work for me for some reason. No errors in the log, and the texture pack loaded like normal. I took a stab at writing my own shader and it ended up working (it also removes the advanced tooltip text). I can pass on the files if you would like them.

TheIdealMatrix commented:
> 



> Thanks! Unfortunately, it didn't work for me for some reason. No errors in the log, and the texture pack loaded like normal. I took a stab at writing my own shader and it ended up working (it also removes the advanced tooltip text). I can pass on the files if you would like them.

You can send them to me Via Discord: `matrixsystempvp`. Then I'll take a look and can add it to this PR if it works.

TheIdealMatrix committed:
Fixed code not working on Apple M2 & code refactor

TheIdealMatrix commented:
@kellenbelle and I fixed the Code to also work on Apple M2.
I also refactored the code a bit, to make it easier to distinguish between minecrafts code and our custom code (good if minecraft changes the code in later Versions)

entonoire commented:
i'm using it with sodium in 1.18.2 and it doesn't work (keep displaying a little bar at the botton of the screen)

kelllllen commented:
> i'm using it with sodium in 1.18.2 and it doesn't work (keep displaying a little bar at the botton of the screen)

This is for 1.20.1 (possibly 1.20.0). You can try to use the original code (not this commit), and if that doesn’t work because you are Mac, you can use our code and change the name from rendertype_gui to position_color, and that most likely will work.

entonoire commented:
unfortunately the original code gives this result and your code doesn't make anything (even by changing the file name)
![image](https://github.com/lolgeny/item-tooltip-remover/assets/43643450/2fcd4a22-8eb9-4f63-bf01-724744d2cb8b)

entonoire commented:
fixed it by adding that (removing those two colors if they are at the right position)

`
if (position.y < -0.96) {
    	if (color.r == 16.0/255 && color.g == 0.0 && color.b == 16.0/255) {
	    	discard;
	    }

	    if (color.r == 40.0/255 && color.g == 0.0 && color.b == 127.0/255) {
	    	discard;
	    }
    }
`
